### PR TITLE
Fix/currency token lock replacement

### PR DESCRIPTION
**Pull request by scasplte2**
## Summary
Adds comprehensive E2E edge case tests for the token lock replacement + delegated stake interaction.

Changes

Core Feature

• Remove currency token lock replacement restriction - Currency token locks can now be replaced (previously only DAG locks supported replacement)
• Fix globalActiveTokenLocksByRef lookup - Include currency token locks from currency snapshots in the replacement lookup map (fixes "NothingToReplace" errors for currency locks)
• Allow same-currency replacement - DAG→DAG and currency→same-currency replacements now work correctly
Files Changed

• ContextualTokenLockValidator.scala - Remove ReplacementIsNotSupported check for currency locks
• TokenLockValidator.scala - Same removal
• TokenLockStateManager.scala - Update filter to allow same-currency replacement
• GlobalSnapshotAcceptanceManager.scala - Include currency locks in globalActiveTokenLocksByRef
Test Metagraph Updates

• Add UsageUpdateWithTokenLockReplacement data type for testing replacement flow
• Update Combiners.scala and LifecycleSharedFunctions.scala to handle new type
CI Test Coverage

New E2E test workflow (test-token-lock-replacement.yml) covering:

1. Replace with same amount → ReplacementLowerThanCurrentTokenLock ❌
2. Replace with less amount → ReplacementLowerThanCurrentTokenLock ❌
3. Replace non-existent ref → NothingToReplace ❌
4. Replace with minimum increase (+1 datum) → ✅
5. Multiple sequential replacements (3x) → ✅
6. Replace while stake in withdrawal → ✅
Testing

• All edge case tests pass on local cluster
• Verified delegated stake currentTokenLockRef and currentAmount update correctly after replacement
Notes

The e2e-test/ directory contains local debugging scripts used during development. These are not part of CI but may be useful for manual testing.

### CODE REVIEW

File: modules/shared/src/test/scala/io/constellationnetwork/json/BrotliCompareApp.scala [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import eu.timepit.refined.types.numeric.{NonNegLong, PosLong}
+import io.circe.Printer
+import io.circe.syntax._
+
+/**
+ * Test app to compare brotli serialization between Tessellation and dag4.js
+ * 
+ * Run with: sbt "shared/test:runMain io.constellationnetwork.json.BrotliCompareApp"
+ */
+object BrotliCompareApp extends IOApp.Simple {
```
Comment (scasplte2) [inline modules/shared/src/test/scala/io/constellationnetwork/json/BrotliCompareApp.scala]: just a test app that I thought might be useful to keep as a comparison but we can remove if y'all think it is clutter

**Comment by scasplte2** [comment]
Currency token locks are disabled until an adequate signalling mechanism is derived to allow metagraph state to react to token lock replacements
